backend/src/main/java/de/sambam/gamenighttracker/service/UserService.java

```diff
@@ -26,4 +28,19 @@ public UserService(UserDb userDb) {
         }
         return Optional.empty();
     }
+
+    public Optional<List<GameSession>> listAllSessions(String id) {
+        Optional<User> user = userDb.findById(id);
+        if (user.isPresent()) {
+            List<Game> gameList = user.get().getPlayedGames();
+            List<GameSession> sessionList = new ArrayList<>();
+            for (Game game : gameList) {
+                for (GameSession session : game.getGameSessionList()) {
+                    sessionList.add(session);
+                }
+            }
```

Comment:
i would use streams here
something like: gameList.stream().flatMap(game -> game.getGameSessionList()).collect(Collectors.toList());

https://www.baeldung.com/java-difference-map-and-flatmap